src/com/facebook/buck/jvm/java/JdkProvidedInMemoryJavac.java

```diff
@@ -16,22 +16,16 @@
 
 package com.facebook.buck.jvm.java;
 
+import com.facebook.buck.jvm.java.javax.SynchronizedToolProvider;
 import com.facebook.buck.rules.SourcePathResolver;
 import com.facebook.buck.util.HumanReadableException;
 import javax.tools.JavaCompiler;
-import javax.tools.ToolProvider;
 
 public class JdkProvidedInMemoryJavac extends Jsr199Javac {
   @Override
   protected JavaCompiler createCompiler(
       JavacExecutionContext context, SourcePathResolver pathResolver) {
-    JavaCompiler compiler;
-    synchronized (ToolProvider.class) {
-      // ToolProvider has no synchronization internally, so if we don't synchronize from the
```

Comment:
I think it would make sense to move it to `SynchronizedToolProvider` as it is not exactly obvious.

Reply:
I didn't get exactly what do you mean, could you explain better, @dmitry-zaitsev ?

Reply:
This synchronized block was moved into `SynchronizedToolProvider`

```
synchronized (ToolProvider.class) {
     compiler = ToolProvider.getSystemJavaCompiler();
}
```

It is not really obvious why this synchronization is performed, so there was a comment left to explain that. Now that this logic is moved into a new class the comment is removed and the purpose of `SynchronizedToolProvider` is not documented anywhere. Would be great to move the contents of this comment into the JavaDoc of `SynchronizedToolProvider`.

Reply:
Done 👍